pyannote/audio/core/model.py

```diff
@@ -217,9 +217,16 @@ def __example_output(
             self.specifications, __example_output, example_output
         )
 
+    def prepare_data(self):
+        self.task.prepare_data()
+
     def setup(self, stage=None):
         if stage == "fit":
-            self.task.setup_metadata()
+            # let the task know about the trainer (e.g for broadcasting)
+            self.task.trainer = self.trainer
+
+        if self.task:
```

Comment:
Maybe add a comment explaining when this happens (fit or validate?) and when this does not (basic inference?)